### ISSUE TITLE
CrossEntropy Loss is more of a regression loss function

### ISSUE THREAD

**Issue by kmedved** [catboost/catboost#2246]
Hello - as I understand it, Catboost currently only supports CrossEntropy loss for CatboostClassifier. This is somewhat contrary to how other state-of-the-art gradient boosted decision trees operate (e.g., LightGBM takes cross_entropy loss as a regression loss function). 

However, this has practical implications as well, namely that CatboostClassifier with CrossEntropy loss isn't really compatible with the bulk of the scikit-learn ecosystem, since you need to pass a continuous label column to it, while scikit-learn is expecting discrete labels in a classifier. For instance, it's impossible to use CatboostClassifier with CrossEntropy loss and BaggingClassifier, since it wants discrete labels.

This would be solved by moving CrossEntropy loss to CatboostRegressor. Would this be possible?

**Comment by ek-ak** [comment]
Hello!
There is two available loss functions for CatboostClassifier: Logloss (works only with  discrete labels) and CrossEntropy (supposed to be used with probabilities and correspondingly works with continuous labels). 
Anyway BaggingClassifier itself needs discrete labels and these is a problem.

**Comment by kmedved** [comment]
Thanks. To clarify, the issue is I'd like to use CrossEntropy loss on continuous labels, so CatBoostClassifier with CrossEntropy loss works for that. However, CatBoostClassifier with CrossEntropy loss cannot be used with either BaggingClassifier or BaggingRegressor. The easiest answer would be to make CrossEntropy loss available as a regression objective in CatboostRegressor, which could then be used with BaggingRegressor. This is how LightGBM handles CrossEntropy loss (reflecting that CrossEntropy takes continuous labels).

To give an example, if I want to use Catboost to predict the probability that it rains on a given day, and then also use Bagging, there is currently no way to do that with the current implementation. Adding CrossEntropy as a regression objective would solve this.

Thanks!

**Comment by ek-ak** [comment]
Catboost Regression class doesn't return probabilities but continious predictions, I mean results not  in [0, 1] interval.

I think you can use simple wrapped  CatBoostClassifier. 
```
from catboost import CatBoostClassifier
from sklearn.datasets import make_classification
from sklearn.ensemble import BaggingClassifier, BaggingRegressor


X, y = make_classification(n_samples=1000, n_features=10)

classifier_bagging = BaggingClassifier(CatBoostClassifier(loss_function='CrossEntropy', iterations=60, verbose=False)).fit(X, y)
classifier_bagging.predict(X)
proba_y = classifier_bagging.predict_proba(X)[:, 1]

class CatBoostClassifierWrapper(CatBoostClassifier):
    def predict(self, *args, **kwargs):
        return self.predict_proba( *args, **kwargs)[:, 0]

regressor_bagging = BaggingRegressor(CatBoostClassifierWrapper(loss_function='CrossEntropy', iterations=60, verbose=False)).fit(X, proba_y)
regressor_bagging.predict(X)
```
And CatBoostClassifier with CrossEntropy loss works with  BaggingClassifier.
